src/test/java/selenium/TableSortAndSearchTest.java

```diff
@@ -0,0 +1,170 @@
+package selenium;
+
+import lombok.extern.slf4j.Slf4j;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import testbase.TestBase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static constants.ByConstants.NEXT_PAGE_BUTTON;
+import static constants.ByConstants.SHOW_ENTRIES_DROPDOWN;
+import static constants.TestConstants.TABLE_TEST_URL;
+
+@Slf4j
+public class TableSortAndSearchTest extends TestBase {
+
+    @Test
+    public void TestTableSortAndSearch() {
+        log.info("Open the URL: {}", TABLE_TEST_URL);
+        driver().get(TABLE_TEST_URL);
+
+        Select showEntriesDropdown = new Select(driver().findElement(SHOW_ENTRIES_DROPDOWN));
+        String optionToBeSelected = "10";
+        String selectedOption = showEntriesDropdown.getFirstSelectedOption().getText();
+        log.info("Set up the Show {} entries filter", optionToBeSelected);
+        if (!selectedOption.equals(optionToBeSelected)) {
+            showEntriesDropdown.selectByVisibleText(optionToBeSelected);
+        }
+        Assert.assertEquals(optionToBeSelected, showEntriesDropdown.getFirstSelectedOption().getText());
+
+        int age = 40;
+        double salary = 150000.0;
+        log.info("Sort employees by following criteria: age > {} and salary <= {}", age, salary);
+        List<Employee> sortedList = new Employee().sortEmployees(age, salary);
+        Assert.assertFalse(sortedList.isEmpty());
+
+        log.info("Print employees who have age > {} and salary <= {}", age, salary);
+        for (Employee employee : sortedList) {
+            System.out.println(employee.toString());
+        }
+    }
+
+    public class Employee {
```

Comment:
1. let's use lombok here
2. move this class to src/main/java/model package (if you don't have such - create this one)
3. getEmployeeDataFromPage, getEmployeeDataFromAllPages, sortEmployees inside of this class violate single-responsibility principle - move those methods to a separate class like Employees

Reply:
Done!